Source/Dafny/SccGraph.cs

```diff
@@ -130,6 +148,13 @@ public class Vertex {
       return GetSCCRepr(n).N;
     }
 
+    /// <summary>
+    /// Idempotently adds 'n' as a vertex.
+    /// </summary>
+    public int GetSCCRepresentativePredecessorCount(Node n) {
+      return GetSCCRepr(n).SccPredecessorCount;
```

Comment:
Has `.SccPredecessorCount` been computed when `GetSCCRepresentativePredecessorCount` is called? If this is up to the caller, say so in the comment. (And if so, and if the caller thinks that the predecessor counts have indeed been computed, then it seems strange to call out that this method idempotently adds the vertex. This means that the `.SccPredecessorCount` is known to be correct for a newly added node. Is that guaranteed?)